src/provider/provider_tracking.c

```diff
@@ -429,6 +466,132 @@ static const char *trackingName(void *provider) {
     return umfMemoryProviderGetName(p->hUpstream);
 }
 
+static umf_result_t trackingGetIpcHandleSize(void *provider, size_t *size) {
+    umf_tracking_memory_provider_t *p =
+        (umf_tracking_memory_provider_t *)provider;
+    return umfMemoryProviderGetIPCHandleSize(p->hUpstream, size);
+}
+
+static umf_result_t trackingGetIpcHandle(void *provider, const void *ptr,
+                                         size_t size, void *ipcData) {
+    umf_tracking_memory_provider_t *p =
+        (umf_tracking_memory_provider_t *)provider;
+    umf_result_t ret = UMF_RESULT_SUCCESS;
+    size_t ipcDataSize = 0;
+    int cached = 0;
+    do {
+        void *value = critnib_get(p->ipcCache, (uintptr_t)ptr);
+        if (value) { //cache hit
```

Comment:
Should we validate the size of the allocation stored in the cache here?

Reply:
I did some code cleanup, please review.